emsdk.py

```diff
@@ -2510,13 +2510,24 @@ def adjusted_path(tools_to_activate, system=False):
     existing_path = os.environ['PATH'].split(ENVPATH_SEPARATOR)
   emsdk_root_path = to_unix_path(emsdk_path())
 
-  existing_emsdk_tools = [i for i in existing_path if to_unix_path(i).startswith(emsdk_root_path)]
-  new_emsdk_tools = [i for i in path_add if not normalized_contains(existing_emsdk_tools, i)]
+  existing_emsdk_tools = []
+  existing_nonemsdk_path = []
+  for entry in existing_path:
+    if to_unix_path(entry).startswith(emsdk_root_path):
+      existing_emsdk_tools.append(entry)
+    else:
+      existing_nonemsdk_path.append(entry)
+
+  new_emsdk_tools = []
+  kept_emsdk_tools = []
+  for entry in path_add:
+    if not normalized_contains(existing_emsdk_tools, entry):
+      new_emsdk_tools.append(entry)
+    else:
+      kept_emsdk_tools.append(entry)
+
+  whole_path = unique_items(new_emsdk_tools + kept_emsdk_tools + existing_nonemsdk_path)
```

Comment:
Can you revert this part of the fix and it will still work?

Could we move this to its own PR?     I'm not against this part but it seems like a refactor that is independent of the actual fix maybe?

Reply:
I would prefer not to. These changes fix the `--permanent` flag, and without this, I am not certain that #634

Additionally, we have already discussed/justified/reviewed the changes here. I do not want to redo all of these just to make these two separate PRs.

Reply:
The reason I'm asking you to do this is because this change seem separate orthogonal to the actually fix being made.

If you revert this block all the tests, including the new ones will still pass right?   Or am I wrong about that?   If I'm right, I'd really like to see this change land without this block.. and this we can land this block as a separate NFC (Non-functional change) cleanup later.

Reply:
It is not orthogonal. That fixes `--permanent`. When I made these changes, my local tests started to work properly. Before that, parts of my PATH were getting removed.

This PR fixes both `--permanent` and `--system`.

Reply:
I just don't see how the old version of this could ever remove anything from `existing_path`... what am I missed?  Where is the bug in the existing code?

Are you saying that if you revert this block one of the tests will start failing?   If I could see which test would fail it might really help me understand where the bugfix is in this block of code.

I'm not trying to be antagonistic, and I just don't want to land something where I don't understand where the original problem is.

As far as I can tell both these block start out with `existing_path` and include all elements of it that are not part of emsdk... how can the old code drop elements from `existing_path`?

Reply:
That's fine. You have proper write access to my branch.

I have a local backup that finally allows me to spend my time writing the WASM code rather than meticulously proving/disproving a PATH editing algorithm.